src/modules/service-catalog/components/service-catalog-item/CollapsibleDescription.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import { Button } from "@zendeskgarden/react-buttons";
+import ChevronUp from "@zendeskgarden/svg-icons/src/16/chevron-up-fill.svg";
+import ChevronDown from "@zendeskgarden/svg-icons/src/16/chevron-down-fill.svg";
+import { useTranslation } from "react-i18next";
+import { getColorV8 } from "@zendeskgarden/react-theming";
+import { XXXL } from "@zendeskgarden/react-typography";
+
+const DescriptionWrapper = styled.div`
+  border-bottom: ${(props) => props.theme.borders.sm}
+    ${(props) => getColorV8("grey", 300, props.theme)};
+  padding-bottom: ${(props) => props.theme.space.lg};
+  margin-right: ${(props) => props.theme.space.xl};
+
+  @media (max-width: ${(props) => props.theme.breakpoints.md}) {
+    margin-right: 0;
+  }
+`;
+
+const ItemTitle = styled(XXXL)`
+  font-weight: ${(props) => props.theme.fontWeights.semibold};
+`;
+
+const CollapsibleText = styled.div<{ expanded: boolean }>`
+  font-size: ${(props) => props.theme.fontSizes.md};
+  text-align: left;
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: ${(props) => (props.expanded ? "none" : 3)};
+  overflow: hidden;
+  margin-top: ${(props) => props.theme.space.md};
+  padding-right: ${(props) => props.theme.space.xl};
+
+  @media (max-width: ${(props) => props.theme.breakpoints.md}) {
+    padding-right: 0;
+  }
+`;
+
+const ToggleButton = styled(Button)`
+  margin-top: ${(props) => props.theme.space.sm};
+  font-size: ${(props) => props.theme.fontSizes.md};
+  &:hover {
+    text-decoration: none;
+  }
+`;
+
+interface CollapsibleDescriptionProps {
+  title: string;
+  description: string;
+}
+
+export const CollapsibleDescription = ({
+  title,
+  description,
+}: CollapsibleDescriptionProps) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [showToggleButton, setShowToggleButton] = useState<boolean>(false);
+  const descriptionRef = useRef<HTMLDivElement>(null);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (descriptionRef.current) {
+      const lineHeight = parseFloat(
+        getComputedStyle(descriptionRef.current).lineHeight
+      );
+      const maxHeight = lineHeight * 3;
+      if (descriptionRef.current.scrollHeight > maxHeight) {
+        setShowToggleButton(true);
+      } else {
+        setShowToggleButton(false);
+      }
+    }
+  }, [description]);
```

Comment:
Would it be possible to do handle the toggle based on the `length` of the description instead of the height?

Reply:
I changed it to length but I'm not sure if the `DESCRIPTION_LENGTH_THRESHOLD = 250` is ok

Reply:
We can revise it with design later with design 👍🏼

Reply:
But it might differ, letters have different width so it might be more characters but with letters that are thin or the other way around. So I don't think description length based on characters count is not good.

Reply:
Yeah I know, this is why I think we should set a higher value, or eventually go back to the previous approach.

If we set a higher value, let's say for example 350 characters, it may happen that we will have like 5 rows and we don't show the collapsible button. I don't think it is a big deal if we don't always show exactly 3 lines. 
The important thing is that we find a good enough threshold to avoid showing the button when there is no more content to show, and this can be tricky especially considering that we will going to support other languages with different characters sets in the future.

On the other hand, the previous approach is more complex, but if it is the only reliable one maybe we should use it. wdyt @luis-almeida?

Reply:
We're not using a monospace font so I don't think it's an issue that the rows we end up with might differ a bit.
It's considerably simpler to do the check based on the based on the `description.length` so I'd go with that approach. We could also just have it in a one-liner without the need for state and effect as mentioned in the other comment 👍🏼

Reply:
We can try `270` 👍🏼  I think this is something we can always change entirely if it doesn't work.
But I think still worth it to try start with the simpler approach.